data_hub_api/docmaps/provider.py

```diff
@@ -194,85 +201,77 @@ def get_single_actions_value_for_preprint_peer_reviewed_step(
+        hypothesis_id = evaluation['hypothesis_id']
+        annotation_created_timestamp = evaluation['annotation_created_timestamp']
+        tags = evaluation['tags']
+        yield get_single_actions_value_for_preprint_peer_reviewed_step(
+            query_result_item=query_result_item,
+            hypothesis_id=hypothesis_id,
+            annotation_created_timestamp=annotation_created_timestamp,
+            tags=tags
         )
 
 
-def get_docmap_actions_value_from_query_result(
-    step_number: int,
-    query_result_item: dict
-) -> list:
-    return list(iter_single_actions_value_from_query_result(
-        step_number,
-        query_result_item
-    ))
+def get_docmaps_step_for_peer_reviewed_status(
+    query_result_item
+):
+    return {
+        'actions': list(iter_single_actions_value_from_query_result(
+            query_result_item=query_result_item
+            )
+        ),
+        'assertions': get_docmap_assertions_value_for_preprint_peer_reviewed_step(
+            query_result_item=query_result_item
+        ),
+        'inputs': get_docmap_inputs_value_for_review_steps(
+            query_result_item=query_result_item
+        )
+    }
 
 
 def generate_docmap_steps(number_of_steps: int, query_result_item: dict) -> dict:
     step_number = 0
     steps_dict = {}
     while step_number < number_of_steps:
         LOGGER.debug('step_number: %r', step_number)
-        step_dict = {
-            'actions': get_docmap_actions_value_from_query_result(
-                step_number,
-                query_result_item
-            ),
-            'assertions': get_docmap_assertions_value_from_query_result(
-                step_number,
-                query_result_item
-            ),
-            'inputs': get_docmap_inputs_value_from_query_result(
-                step_number,
-                query_result_item
-            ),
+        step_ranking_dict = {
             'next-step': (
                 '_:b' + str(step_number + 1) if step_number + 1 < number_of_steps else None
             ),
             'previous-step': '_:b' + str(step_number - 1) if step_number > 0 else None
         }
-
-        steps_dict['_:b'+str(step_number)] = step_dict
-
+        if step_number == 0:  # manuscript-published
+            step_dict = get_docmaps_step_for_manuscript_published_status(
+                query_result_item=query_result_item
+            )
+        elif step_number == 1:  # under-review
+            step_dict = get_docmaps_step_for_under_review_status(
+                query_result_item=query_result_item
+            )
+        elif step_number == 2:  # peer-reviewed
+            step_dict = get_docmaps_step_for_peer_reviewed_status(
+                query_result_item=query_result_item
+            )
+        steps_dict['_:b'+str(step_number)] = dict(step_dict, **step_ranking_dict)
         step_number += 1
     return remove_key_with_none_value_only(steps_dict)
 
 
+def get_docmap_step(query_result_item: dict) -> int:
```

Comment:
This should probably be `get_docmap_step_count` or `get_docmap_step_count_for_query_result_item`?

Reply:
I still believe that tying it to the count of steps makes it a less readable as the step index becomes a magic number.

Instead something simple like this would communicate clearly what steps it generates and under what condition:

```python
def iter_docmap_steps_for_query_result_item(query_result_item: dict) -> Iterable[dict]:
    yield get_manuscript_published_step_for_query_result_item(query_result_item)
    yield get_under_review_step_for_query_result_item(query_result_item)
    if not query_result_item['evaluations']:
        yield get_peer_reviewed_step_for_query_result_item(query_result_item)
```

Reply:
created seperate issue for this